client/src/components/Reports/index.js

```diff
@@ -1,16 +1,179 @@
+import TableRow from "@material-ui/core/TableRow";
+import Paper from "@material-ui/core/Paper";
+import Button from "@material-ui/core/Button";
+import red from "@material-ui/core/colors/red";
+import InputLabel from '@material-ui/core/InputLabel';
+import FormControl from '@material-ui/core/FormControl';
+import FormHelperText from '@material-ui/core/FormHelperText';
+import NativeSelect from '@material-ui/core/NativeSelect';
+import Input from '@material-ui/core/Input';
+import { getData } from "../../utils/getData";
 
-class Reports extends Component {
+
+const styles = theme => ({
+  root: {
+    width: "100%",
+    marginTop: theme.spacing.unit * 3,
+    overflowX: "auto",
+    display: 'flex',
+  flexWrap: 'wrap',
+  },
+  table: {
+    minWidth: 700
+  },
+  button:{
+   marginLeft:"300%",
+   marginTop:"20px",
+   backgroundColor:red[900],
+ },
+ title:{
+   color:red[900]
+ },
+ head:{
+   backgroundColor:red[200],
+   color: theme.palette.common.red,
+ },
+ primary: { main: red[500] },
+ formControl: {
+    margin: theme.spacing.unit,
+    minWidth: 120,
+    marginLeft:"20%"
+  },
+  selectEmpty: {
+    marginTop: theme.spacing.unit * 2,
+  },
+});
+
+class Reports extends React.Component {
+  state = {
+   month: new Date().getMonth() + 1,
+   year:new Date().getFullYear(),
+   labelWidth: 0,
+   carLogs:null
+ };
+
+  componentDidMount() {
+    this.getLogs();
+  }
+  handleChange = name => event => {
+   this.setState({ [name]: event.target.value },()=>{
+     this.getLogs();
+   });
+ };
+ getLogs = () => {
+   getData(`/logs/${this.state.year}/${this.state.month}`).then(carLogs => {
+     this.setState({ carLogs });
+   });
+ };
+
+  download=()=>{
+    console.log('year',this.state.year,'month',this.state.month);
+    fetch(`/logmonth/${this.state.year}/${this.state.month}`).then(response=>{
+      return response.blob();
+    }).then(blob=>{
+      download(blob,"logs.xlsx")
+    })
+  }
   render() {
+    const { classes } = this.props;
+    const { carLogs } = this.state;
     return (
-      <React.Fragment>
-        <Add />
-        <Show />
-      </React.Fragment>
-    );
+      <div>
+      <h1 align="center" className={classes.title}>Vehicle Log Sheet</h1>
+
+      <FormControl className={classes.formControl}>
+          <InputLabel htmlFor="uncontrolled-native">Month</InputLabel>
+          <NativeSelect defaultValue={this.state.month}
+          input={<Input name="month"
+          id="uncontrolled-native"
+          onChange={this.handleChange('month')}
+           />}>
+            <option value="" />
+            <option value={1}>1</option>
+            <option value={2}>2</option>
+            <option value={3}>3</option>
+            <option value={4}>4</option>
+            <option value={5}>5</option>
+            <option value={6}>6</option>
+            <option value={7}>7</option>
+            <option value={8}>8</option>
+            <option value={9}>9</option>
```

Comment:
this repetitive code could be replacted by `new Array(12).fill(0).map((_,index)=>{
 return <option value {index+1)>index+1</option>`

Reply:
Done :+1: